app/shared/api-services/poll/poll-factory.js

```diff
@@ -3,77 +3,125 @@
 angular.module('apiServices.poll.factory', [
   'apiServices',
   'apiServices.mixins.fieldDisplay',
+  'bulbsCmsApp.nonRestmodListPage',
   'filters.moment',
   'lodash'
 ])
-.factory('Poll', ['$filter', '$http', '$q', '_', 'moment', function ($filter, $http, $q, _, moment) {
-
-  var filter;
-  var pollInfo;
-  var pollUrl = '/cms/api/v1/poll/';
+.factory('Poll',
+  ['$filter', '$http', '$q', '_', 'moment', 'Utils',
+  function ($filter, $http, $q, _, moment, Utils) {
 
   var error = function(message) {
     return new Error('Poll Error: ' + message);
   };
 
-  function getPoll(pollId) {
-    filter = $filter('date_string_to_moment');
+  var fields = [{
+      title: 'Poll Name',
+      sorts: 'title',
+    }, {
+      title: 'Publish Date',
+      sorts: 'publish_date',
+      content: function (poll) {
+        return poll.published ? poll.published.format('MM/DD/YY h:mma') : '—';
+      },
+    }, {
+      title: 'Close Date',
+      sorts: 'end_date',
+      content: function (poll) {
+        return poll.end_date ? poll.end_date.format('MM/DD/YY h:mma') : '—';
+      },
+  }];
+  var name = 'Poll';
+  var namePlural = 'Polls';
+  var pollUrl = '/cms/api/v1/poll/';
 
-    return $http.get(pollUrl + pollId)
-    .then(function (response) {
-      response.data.end_date = filter(response.data.end_date);
-      return response.data;
-    });
+  function parsePayload (payload) {
+    var data = _.clone(payload);
+    var filter = $filter('date_string_to_moment');
+    data.end_date = filter(data.end_date);
+    data.published = filter(data.published);
+    return data;
   }
 
-  function postPoll(data) {
-    if(_.isUndefined(data.title) && _.isUndefined(data.question_text)) {
+  function cleanPayload (payload) {
+    var momentToDateString = $filter('moment_to_date_string');
+    payload = _.clone(payload);
```

Comment:
Store the clone in a differently named variable so reading this function is a bit clearer.